generate/adapter.py

```diff
@@ -77,15 +93,15 @@ def main(
 
     fabric.print(f"Loading model {str(checkpoint_path)!r} with {config.__dict__}", file=sys.stderr)
     t0 = time.perf_counter()
-    with fabric.init_module(empty_init=True), quantization(quantize):
+    with fabric.init_module(empty_init=True), gptq_quantization(quantize == "gptq.int4"):
         model = GPT(config)
     fabric.print(f"Time to instantiate model: {time.perf_counter() - t0:.02f} seconds.", file=sys.stderr)
 
     t0 = time.perf_counter()
     checkpoint = lazy_load(checkpoint_path)
     adapter_checkpoint = lazy_load(adapter_path)
     checkpoint.update(adapter_checkpoint.get("model", adapter_checkpoint))
-    model.load_state_dict(checkpoint, strict=quantize is None)
```

Comment:
Is this resolved by the new state dict hooks we added?

Reply:
Yes, exactyl